theano/gof/cmodule.py

```diff
@@ -1787,6 +1787,28 @@ def _try_flags(flag_list, preambule="", body="",
                                          flags=flag_list, try_run=try_run,
                                          output=output, compiler=compiler)
 
+def try_march_flag(flags):
+    import textwrap
+    test_code = textwrap.dedent("""\
+            #include <cmath>
+            int main(int argc, char** argv)
+            {
+                float Nx = 5.0;
+                int Sx = 25.0;
+                double r = Nx + sqrt(Sx);
+                if ((r - 10.) > 1e-6 || (r - 10.) < -1e-6)
+                {
+                    return -1;
+                }
+                return 0;
+            }
+            """)
```

Comment:
the c test code  should correspond to one of the use case define in the ticket http://github.com/Theano/Theano/issues/4002